src/screens/ExchangeModal.js

```diff
@@ -139,6 +139,31 @@ export default function ExchangeModal({
     addListener,
   } = useNavigation();
 
+  const defaultOutputAssetOverride = useMemo(() => {
+    if (defaultInputAsset && defaultOutputAsset) {
+      if (
+        defaultInputAsset.type !== defaultOutputAsset.type &&
+        defaultOutputAsset?.implementations?.[defaultInputAsset?.type]?.address
+      ) {
+        if (defaultInputAsset.type !== Network.mainnet)
+          defaultOutputAsset.mainnet_address = defaultOutputAsset.address;
+
+        defaultOutputAsset.address =
+          defaultOutputAsset.implementations[defaultInputAsset?.type].address;
+        defaultOutputAsset.type = defaultInputAsset.type;
+        defaultOutputAsset.uniqueId =
+          defaultOutputAsset.type === Network.mainnet
+            ? defaultOutputAsset?.address
+            : `${defaultOutputAsset?.address}_${defaultOutputAsset?.type}`;
+        return defaultOutputAsset;
+      } else {
+        return null;
+      }
+    } else {
+      return defaultOutputAsset;
+    }
+  }, [defaultInputAsset, defaultOutputAsset]);
```

Comment:
this is a bit confusing, are you overriding params inside an object with other params of the same object? maybe a comment explaining could be nice

Reply:
updated to use a new object and added a comment to explain the logic